pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -256,11 +257,13 @@ func (d *DependenciesDistributor) OnResourceBindingUpdate(oldObj, newObj interfa
 	}
 
 	if oldBindingObject.Generation == newBindingObject.Generation {
+		klog.V(4).Infof("Dropping resource binding(%s/%s) as the Generation is not changed.", newBindingObject.Namespace, newBindingObject.Name)
 		return
 	}
 
-	// prevent newBindingObject from the queue if it's not in Scheduled condition
-	if !helper.IsBindingScheduled(&newBindingObject.Status) {
+	// prevent newBindingObject from the queue if it's not scheduled yet.
+	if len(oldBindingObject.Spec.Clusters) == 0 && len(newBindingObject.Spec.Clusters) == 0 {
```

Comment:
When `len(oldBindingObject.Spec.Clusters) == 0` and `len(newBindingObject.Spec.Clusters) != 0`, does it also need to be added to the queue?